plugins/Ecommerce/Columns/ProductViewCategory.php

```diff
@@ -0,0 +1,133 @@
+    protected $categoryNumber = 1;
+
+    public function configureSegments(SegmentsList $segmentsList, DimensionSegmentFactory $dimensionSegmentFactory)
+    {
+        $individualProductCategorySegments = $this->getProductCategorySegments(ProductCategory::PRODUCT_CATEGORY_COUNT);
+
+        // add individual productCategoryN segments for use as a union (these segments are not available through the UI/API)
+        foreach ($individualProductCategorySegments as $i => $productCategoryName) {
+            $productCategoryColumnName = 'idaction_product_cat';
+            if ($i > 0) {
+                $productCategoryColumnName .= $i + 1;
+            }
+
+            $segment = new Segment();
+            $segment->setCategory($this->category);
+            $segment->setType('dimension');
+            $segment->setName($this->getName() . ' ' . ($i + 1));
+            $segment->setSegment($productCategoryName);
+            $segment->setSqlFilter('\\Piwik\\Tracker\\TableLogAction::getIdActionFromSegment');
+            $segment->setSqlSegment('log_link_visit_action.' . $productCategoryColumnName);
+            $segment->setIsInternal(true);
+            $segment->setSuggestedValuesCallback(function ($idSite, $maxValuesToReturn, $table) {
+                $values = [];
+                foreach ($table->getRows() as $row) {
+                    foreach ($row->getColumn('actionDetails') as $actionRow) {
+                        if (isset($actionRow['productViewCategories'])) {
+                            $values = array_merge($values, $actionRow['productViewCategories']);
+                        }
+                    }
+                }
+                return $values;
+            });
+            $segmentsList->addSegment($dimensionSegmentFactory->createSegment($segment));
+        }
+
+        // add a union of these individual columns as productCategory
+        $segment = new Segment();
+        $segment->setCategory($this->category);
+        $segment->setType('dimension');
+        $segment->setSegment('productViewCategory');
+        $segment->setName($this->getName());
+        $segment->setUnionOfSegments($individualProductCategorySegments);
+        $segmentsList->addSegment($dimensionSegmentFactory->createSegment($segment));
+    }
+
+    private function getProductCategorySegments($categoryCount)
+    {
+        $result = [];
+        for ($i = 0; $i < $categoryCount; ++$i) {
+            $segmentName = 'productViewCategory' . ($i + 1);
+            $result[] = $segmentName;
+        }
+        return $result;
+    }
+
+    public function getDbColumnJoin()
+    {
+        return new ActionNameJoin();
+    }
+
+    public function getDbDiscriminator()
+    {
+        return new Discriminator('log_action', 'type', Action::TYPE_ECOMMERCE_ITEM_CATEGORY);
+    }
+
+    public function onLookupAction(Request $request, Action $action)
+    {
+        $categories = Common::unsanitizeInputValue($request->getParam('_pkc'));
```

Comment:
should maybe only getParam for category if it has the param to not call like unsanitzeInputValue(null)